src/screens/VisualTests/VisualTests.tsx

```diff
@@ -150,104 +93,229 @@ export const VisualTestsWithoutSelectedBuildId = ({
+        if (!userCanReview) throw new Error("No permission to review tests");
+        const { error } = await runMutation({ input });
+        if (error) throw error;
+        onReviewSuccess?.(input);
+      } catch (err) {
+        onReviewError?.(err, input);
+      }
+    },
+    [onReviewSuccess, onReviewError, runMutation, buildIsReviewable, userCanReview]
+  );
+
+  const acceptTest = useCallback(
+    (testId: string, batch: ReviewTestBatch = ReviewTestBatch.Spec) =>
+      reviewTest({ status: ReviewTestInputStatus.Accepted, testId, batch }),
+    [reviewTest]
+  );
+
+  const unacceptTest = useCallback(
+    (testId: string, batch: ReviewTestBatch = ReviewTestBatch.Spec) =>
+      reviewTest({ status: ReviewTestInputStatus.Pending, testId, batch }),
+    [reviewTest]
+  );
+
+  return { isReviewing, acceptTest, unacceptTest, buildIsReviewable, userCanReview };
+};
+
+export const VisualTestsWithoutSelectedBuildId = ({
+  selectedBuildInfo,
+  setSelectedBuildInfo,
+  dismissBuildError,
+  localBuildProgress,
+  startDevBuild,
+  setAccessToken,
+  setOutdated,
+  updateBuildStatus,
+  projectId,
+  gitInfo,
+  storyId,
+}: VisualTestsProps) => {
+  const { addNotification } = useStorybookApi();
+
+  const {
+    hasData,
+    hasProject,
+    hasSelectedBuild,
+    lastBuildOnBranch,
+    lastBuildOnBranchIsReady,
+    lastBuildOnBranchIsSelectable,
+    selectedBuild,
+    selectedBuildMatchesGit,
+    queryError,
+    rerunQuery,
+    userCanReview,
+  } = useBuild({
+    projectId,
+    storyId,
+    gitInfo,
+    selectedBuildInfo,
+  });
+
+  const reviewState = useReview({
+    buildIsReviewable: !!selectedBuild && selectedBuild.id === lastBuildOnBranch?.id,
+    userCanReview,
+    onReviewSuccess: rerunQuery,
+    onReviewError: (err, update) => {
+      if (err instanceof Error) {
+        addNotification({
+          id: "chromatic/errorAccepting",
+          // @ts-expect-error we need a better API for not passing a link
+          link: undefined,
+          content: {
+            headline: `Failed to ${
+              update.status === ReviewTestInputStatus.Accepted ? "accept" : "unaccept"
+            } changes`,
+            subHeadline: err.message,
+          },
+          icon: {
+            name: "cross",
+            color: "red",
+          },
+        });
+      }
+    },
+  });
+
+  // Currently only used by the sidebar button to show a blue dot ("build outdated")
+  useEffect(() => setOutdated(!selectedBuildMatchesGit), [selectedBuildMatchesGit, setOutdated]);
 
   // We always set status to the next build's status, as when we change to a new story we'll see
   // the next builds
-  const testsForStatus =
-    lastBuildOnBranch &&
-    "testsForStatus" in lastBuildOnBranch &&
-    lastBuildOnBranch.testsForStatus &&
-    getFragment(FragmentStatusTestFields, lastBuildOnBranch.testsForStatus.nodes);
+  useEffect(() => {
+    // if (!lastBuildOnBranchIsSelectable) return;
```

Comment:
Why change this behaviour?

Reply:
I didn't. This line wasn't supposed to be here anymore. We actually _do_ want to run the update even if the build isn't selectable (can't be switched to), because the update also clears the existing statuses and we still want that to happen. This is how it worked before too.

Effectively all I've done is move the `testsForStatus` and `testsToStatusUpdate` calculations inside the useEffect.

Reply:
I still don't think you should change behaviour in a refactor PR.

Reply:
I agree, and didn't intend to really change anything here. I hadn't considered the difference in behavior between checking the build vs the update.